resources/Examples/compact_osw/measures/AlfalfaEPlusPython/measure.py

```diff
@@ -0,0 +1,97 @@
+"""insert your copyright here.
+
+# see the URL below for information on how to write OpenStudio measures
+# http://nrel.github.io/OpenStudio-user-documentation/reference/measure_writing_guide/
+"""
+
+import openstudio
+
+
+class AlfalfaEPlusPython(openstudio.measure.EnergyPlusMeasure):
+    """An EnergyPlusMeasure."""
+
+    def name(self):
+        """Returns the human readable name.
+
+        Measure name should be the title case of the class name.
+        The measure name is the first contact a user has with the measure;
+        it is also shared throughout the measure workflow, visible in the OpenStudio Application,
+        PAT, Server Management Consoles, and in output reports.
+        As such, measure names should clearly describe the measure's function,
+        while remaining general in nature
+        """
+        return "AlfalfaEPlusPython"
+
+    def description(self):
+        """Human readable description.
+
+        The measure description is intended for a general audience and should not assume
+        that the reader is familiar with the design and construction practices suggested by the measure.
+        """
+        return "DESCRIPTION_TEXT"
+
+    def modeler_description(self):
+        """Human readable description of modeling approach.
+
+        The modeler description is intended for the energy modeler using the measure.
+        It should explain the measure's intent, and include any requirements about
+        how the baseline model must be set up, major assumptions made by the measure,
+        and relevant citations or references to applicable modeling resources
+        """
+        return "MODELER_DESCRIPTION_TEXT"
+
+    def arguments(self, workspace: openstudio.Workspace):
+        """Prepares user arguments for the measure.
+
+        Measure arguments define which -- if any -- input parameters the user may set before running the measure.
+        """
+        args = openstudio.measure.OSArgumentVector()
+
+        return args
+
+    def run(
+        self,
+        workspace: openstudio.Workspace,
+        runner: openstudio.measure.OSRunner,
+        user_arguments: openstudio.measure.OSArgumentMap,
+    ):
+        """Defines what happens when the measure is run."""
+        super().run(workspace, runner, user_arguments)  # Do **NOT** remove this line
+
+        if not (runner.validateUserArguments(self.arguments(workspace), user_arguments)):
+            return False
+
+        alfalfa = runner.alfalfa()
+
+        # Test Meters
+        alfalfa.exposeMeter("Electricity:Facility", "Electricity Meter String:EPlus:Python")
+
+        meter_object = openstudio.IdfObject.load("Output:Meter, Electricity:Facility;").get()
+        alfalfa.exposeFromObject(meter_object, "Electricity Meter IDF:Eplus:Python")
```

Comment:
Does this method work with a ModelObject ?

Reply:
yes